packages/backend-function/src/schedule_parser.ts

```diff
@@ -0,0 +1,209 @@
+import { CronOptions, Rule, Schedule } from 'aws-cdk-lib/aws-events';
+import { NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs';
+import * as targets from 'aws-cdk-lib/aws-events-targets';
+import { AmplifyUserError } from '@aws-amplify/platform-core';
+import type { Cron, Rate, TimeInterval } from './factory.js';
+
+/**
+ * Parses function schedule props in order to create EventBridge rules.
+ */
+export class ScheduleParser {
+  /**
+   * Initialize EventBridge rules
+   */
+  constructor(
+    private readonly lambda: NodejsFunction,
+    private readonly timeIntervals: TimeInterval[]
+  ) {
+    this.timeIntervals.forEach((interval, index) => {
+      if (isRate(interval)) {
+        const { value, unit } = parseRate(interval);
+
+        if (value && !isPositiveWholeNumber(value)) {
+          throw new Error(`schedule must be set with a positive whole number`);
+        }
+
+        if (
+          value &&
+          lambda.timeout &&
+          unit === 'm' &&
+          value * 60 < lambda.timeout.toSeconds()
+        ) {
+          const timeout = lambda.timeout.toSeconds();
+
+          throw new Error(
+            `schedule must be greater than the timeout of ${timeout} ${
+              timeout > 1 ? 'seconds' : 'second'
+            }`
+          );
+        }
+      } else {
+        if (!isValidCron(interval)) {
+          // TODO: Better error messaging here or throw within each part of isValidCron in order to give more concise error messages
+          throw new Error(`schedule cron expression is not valid`);
+        }
+      }
+
+      try {
+        // Lambda name will be prepended to rule id, so only using index here for uniqueness
+        const rule = new Rule(this.lambda, `lambda-schedule${index}`, {
+          schedule: Schedule.cron(translateToCronOptions(interval)),
+        });
+
+        rule.addTarget(new targets.LambdaFunction(this.lambda));
```

Comment:
do we need to create a new target for each rule or can we reuse the same target since they are all pointing to the same lambda?

Reply:
Good catch, I believe we should be able to create the target once and reuse for each rule.